src/lib/telemetry.rs

```diff
@@ -11,17 +12,42 @@ use tracing_subscriber::{layer::SubscriberExt, EnvFilter, Registry};
 use crate::settings::Settings;
 use crate::version::{read_version, VERSION_FILE};
 
+pub enum TraceType {
+    AicRecordCreate,
+    AicRecordCreateFailed,
+}
+
+impl TraceType {
+    fn as_str(&self) -> &'static str {
+        match self {
```

Comment:
An alternative to this could be to derive `strum_macros::Display` and then applying a `serialize` attribute to each enum value.

https://docs.rs/strum_macros/0.24.0/strum_macros/derive.Display.html